array_experimental_test.go

```diff
@@ -69,11 +69,17 @@ func TestGetConsolidationPlan(t *testing.T) {
 	err = array.Open(TILEDB_WRITE)
 	require.NoError(t, err)
 
+	// Create subarray
+	subarray, err := array.NewSubarray()
```

Comment:
from a usability perspective it seems like NewSubarray could accept accept the parameters SetSubarray does and do the same logic. Do we ever SetSubarray on the same alloced subarray with different params?

Reply:
Calling `SetSubArray` is one of the two options to configure a subarray object; the other is by calling `AddRange(ByName)` for each dimension.

`NewSubarray` could accept the raw values optionally as a shortcut (I don't like it personally but I don't have enough experience to have an opinion on Go API design), but we should better do it on a subsequent PR.